kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -262,8 +257,8 @@ public boolean metadataRequiresFeatureToBeEnabled(Protocol protocol, Metadata me
     }
 
     @Override
-    public boolean hasKernelWriteSupport(Metadata metadata) {
-      return false; // TODO: yet to be implemented in Kernel
+    public Set<TableFeature> requiredFeatures() {
+      return new HashSet<>(Arrays.asList(TableFeatures.VARIANT_RW_FEATURE));
```

Comment:
one issue (not sure if it is):
If the table already has the `varaintType-preview` and you enable shredding through metadata config. With this `requiredFeatures`, it will have both the `variantType` and `variantType-preview` in the table protocol.

Reply:
I think this abstraction makes sense -- if you have a table with variantType-preview, and you somehow (either through implicit updates, or explicitly) enable variantType, can we remove variantType-preview?

Reply:
i don't think theres a concern for now if a small subset of tables have both. Agree it would be cleaner to somehow remove the `-preview` feature, but AFAIK we don't currently have that capability.